### PR TITLE
feat(e2e): `Manifest.Validators` is not a pointer to a map, but a map

### PR DESCRIPTION
**Pull request by cason**
As per title.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by sergio-mena — APPROVED** [review]
🎸

File: test/e2e/pkg/testnet.go [hunk 1]
```diff
@@ -281,14 +281,12 @@ func NewTestnetFromManifest(manifest Manifest, file string, ifd InfrastructureDa
 	}
 
 	// Set up genesis validators. If not specified explicitly, use all validator nodes.
-	if testnet.Validators == nil || len(*testnet.Validators) == 0 {
-		validatorsMap := make(map[string]int64)
```
Comment (alesforz) [inline test/e2e/pkg/testnet.go:285]: given that the `validatorsMap[node.Name] = 100` causes an "assignment to nil map" panic, I think we must keep the  nil check + initialization if needed.

File: test/e2e/runner/setup.go [hunk 2]
```diff
@@ -150,7 +150,7 @@ func MakeGenesis(testnet *e2e.Testnet) (types.GenesisDoc, error) {
 	if testnet.PbtsUpdateHeight == -1 {
 		genesis.ConsensusParams.Feature.PbtsEnableHeight = testnet.PbtsEnableHeight
 	}
-	for valName, power := range *testnet.Manifest.Validators {
+	for valName, power := range testnet.Validators {
```
Comment (alesforz) [inline test/e2e/runner/setup.go:153]: I it correct here to drop one "level" of dereference? From `testnet.Manifest.Validators` to `testnet.Validators`.

File: test/e2e/tests/validator_test.go [hunk 3]
```diff
@@ -153,7 +153,7 @@ type validatorSchedule struct {
 
 func newValidatorSchedule(t *testing.T, testnet *e2e.Testnet) *validatorSchedule {
 	t.Helper()
-	valMap := *testnet.Manifest.Validators        // genesis validators
+	valMap := testnet.Validators                  // genesis validators
```
Comment (alesforz) [inline test/e2e/tests/validator_test.go:156]: Same as [above](https://github.com/cometbft/cometbft/pull/4177/files#r1774903314)

File: test/e2e/runner/setup.go (hunk 2 above)
Comment (cason) [reply test/e2e/runner/setup.go:153]: I think the code is clear. `Testnet` "extends `Manifest` so it inherits its fields.

We are using it like that in the remaining of this method.

File: test/e2e/tests/validator_test.go (hunk 3 above)
Comment (cason) [reply test/e2e/tests/validator_test.go:156]: Same reply as above.

File: test/e2e/runner/setup.go (hunk 2 above)
Comment (alesforz) [reply test/e2e/runner/setup.go:153]: You are right 👍

**Review by alesforz — APPROVED** [review]

**Pull request merged**